src/storage/write_ahead_log.cpp

```diff
@@ -280,8 +280,9 @@ void WriteAheadLog::WriteCreateIndex(const IndexCatalogEntry &entry) {
 
 	// now serialize the index data to the persistent storage and write the index metadata
 	auto &duck_index_entry = entry.Cast<DuckIndexEntry>();
+	auto &table_idx_list = duck_index_entry.GetDataTableInfo().GetIndexes();
+	lock_guard<mutex> lock(table_idx_list.indexes_lock);
 	auto &indexes = duck_index_entry.GetDataTableInfo().GetIndexes().Indexes();
```

Comment:
Can this be rewritten to use `indexes.Scan` instead of exposing the `indexes_lock` internal?